codegen/masm/src/masm/program.rs

```diff
@@ -250,6 +246,15 @@ impl From<&hir::Program> for Program {
     }
 }
 
+impl fmt::Display for Program {
```

Comment:
I think this is fine as a short-term thing, but this is missing the code that will go in the top-level `begin` block of a `Program` (which we're not currently representing here, but will be added very soon, since it is where we will need to place code for initializing memory and things of that nature). We'll also want to be careful that we don't treat this as a canonical representation of a `Program` in textual form, since Miden Assembly doesn't support multiple modules in a single file. I'm assuming we're using this for test assertions though, so for that purpose I think this is fine, just be aware that this format is likely to change, so if we use it in a bunch of test assertions, we'll have to update all of them when that happens (which may well be unavoidable until I get the `filecheck`/`lit` tooling updated).

Reply:
Good catch on the missing entrypoint. Yes, it's for the test assertions only. I don't see how this implementation can be limited to tests, but I can probably switch it to `Debug` trait to emphasize its "internaliness".